core/src/OpenACC/Kokkos_OpenACC_Team.hpp

```diff
@@ -134,17 +141,30 @@ class OpenACCTeamMember {
                     const int team_size,
                     const int vector_length)  // const TeamPolicyInternal<
                                               // OpenACC, Properties ...> & team
-      : m_team_size(team_size),
-        m_league_rank(league_rank),
+      : m_league_rank(league_rank),
         m_league_size(league_size),
+        m_team_size(team_size),
         m_vector_length(vector_length) {
 #ifdef KOKKOS_COMPILER_NVHPC
     m_team_rank = __pgi_vectoridx();
+#else
+#ifdef KOKKOS_COMPILER_CLANG
+    m_team_rank = omp_get_thread_num();
 #else
     m_team_rank = 0;
+#endif
 #endif
   }
 
+  OpenACCTeamMember(const int league_rank, const int league_size,
+                    const int team_rank, const int team_size,
+                    const int vector_length)
```

Comment:
If that constructor is only needed when forcing the most as device I am tempted to guard it's definition since it is potentially confusing to have both weakly-typed constructor coexist.
If applicable hide the other definition for host as device.

Reply:
Disabled unused constructor as suggested.